app/controllers/admin/ride_uploads_controller.rb

```diff
@@ -0,0 +1,139 @@
+class Admin::RideUploadsController < Admin::AdminApplicationController
+  before_action :set_ride_upload, only: [:show, :edit, :update, :schedule, :destroy]
+  before_action :set_ride_zone, only: [:index, :show, :new, :edit, :create, :update, :destroy]
+
+  # GET /ride_uploads
+  def index
+    @ride_uploads = RideUpload.where( ride_zone_id: @ride_zone.id )
+  end
+
+  # GET /admin/ride_zones/ca-25/ride_uploads/
+  def show
+    @failed_potential_rides = @ride_upload.failed_potential_rides
+  end
+
+  # GET /admin/ride_zones/ca-25/ride_uploads/new
+  def new
+    @ride_upload = RideUpload.new
+  end
+
+  # GET /admin/ride_zones/ca-25/ride_uploads/1/edit
+  def edit
+  end
+
+  # POST /admin/ride_zones/ca-25/ride_uploads
+  def create
+    @ride_upload = RideUpload.new(ride_upload_params)
+    @ride_upload.user = current_user
+    @ride_upload.ride_zone = @ride_zone
+    
+    # TODO: Should a radius check be added to PotentialRide, and should it be made geolocatable?
+    if @ride_upload.save
+      row_number = 0
+      total_rows = 0
+      successful_rows = 0
+      CSV.foreach( @ride_upload.csv_on_disk, headers: true ) do |row|
```

Comment:
This block seems like a good thing to put in a helper method or move to a model or utility method. I imagine it would return a simple data structure containing the various row counts.

Reply:
@dryan shit that was an accident, i'd hacked seeds.rb to load data into the production database and didn't mean to commit it. Should have just written a script. Will def remove, I'll sync with you.

Reply:
cool--future PR, if that's cool?

Reply:
SGTM